### PR TITLE
Dispatch files

### PR DESCRIPTION
**Pull request by Soullivaneuh**
- [x] Make diff from files provided on config
- [x] Apply commit if diff on branch
- [x] Make placeholder to replace like `{{ stable_branch }}`, `{{ project_name }}`...

### CODE REVIEW

File: .sonata-project.yml [hunk 1]
```diff
@@ -24,6 +24,10 @@ labels:
   - name: help wanted
     color: '0e8a16'
 
+files:
+  - doc/CONTRIBUTING.md:CONTRIBUTING.md
+  - tools/editorconfig/.editorconfig:.editorconfig
```
Comment (greg0ire) [inline .sonata-project.yml]: Maybe we should keep it simple and have only one tree, matching exactly the destination tree. That way we may simply use a recursive copy.

Comment (Soullivaneuh) [inline .sonata-project.yml]: Yes maybe. Have to test, didn't decide yet.

We also have more complicated file like `.travis.yml` that will needs a twig template...

Comment (greg0ire) [inline .sonata-project.yml]: Puppet separates into a `file` folder and a `templates` directory, but I guess having different extensions for files in the same directory would be enough…

**Comment by Soullivaneuh** [comment]
@greg0ire I updated the code to manage directories. What do you think?

**Comment by greg0ire** [comment]
IMO, if you still need the `files` config entry, there is a problem. One should be able to sync a new file without having to configure anything. In [git template](https://github.com/greg0ire/git_template), I segregated files that are supposed to be rsynced in a separate directory (`template`).

**Comment by Soullivaneuh** [comment]
@greg0ire like this so?

**Comment by greg0ire** [comment]
Exactly :+1:

File: src/Console/Command/DispatchCommand.php [hunk 2]
```diff
@@ -12,28 +12,37 @@
...
 use Symfony\Component\Yaml\Yaml;
 
 /**
  * @author Sullivan Senechal <soullivaneuh@gmail.com>
  */
 class DispatchCommand extends Command
 {
+    const GITHUB_GROUP = 'sonata-project';
+    const GITHUB_USER = 'SonataCI';
+    const GITHUB_EMAIL = 'ci@sonata-project.org';
```
Comment (Soullivaneuh) [inline src/Console/Command/DispatchCommand.php]: @rande the PR is ready. All I need before merge is the real email of `SonataCI` user.

Could you please give me this one?

File: src/Console/Command/DispatchCommand.php [hunk 3]
```diff
@@ -197,4 +236,116 @@ private function updateLabels($repositoryName)
...
+            ->config('user.name', static::GITHUB_USER)
+            ->config('user.email', static::GITHUB_EMAIL)
+        ;
+
+        foreach ($projectConfig['branches'] as $branch => $branchConfig) {
+            $this->io->section('Files for '.$branch);
+
+            $git
+                ->reset(array('hard' => true))
+                ->checkout('-b', $branch, '-t', 'origin/'.$branch)
```
Comment (greg0ire) [inline src/Console/Command/DispatchCommand.php]: Using long options would be more explicit (`--track` instead of `-t`)

**Pull request merged**